mobile/src/bundles/common/components/form-field/form-field.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import {
+    type FieldErrors,
+    type FieldPath,
+    type FieldValues,
+} from 'react-hook-form';
+
+import { Text, View } from '~/bundles/common/components/components';
+import { globalStyles } from '~/bundles/common/styles/global-styles';
+
+import { styles } from './styles';
+
+type Properties<T extends FieldValues> = {
+    errors: FieldErrors<T>;
+    name: FieldPath<T>;
+    children: React.ReactNode;
+    label?: string;
+    required?: boolean;
+    hasError?: boolean;
+};
+
+const FormField = <T extends FieldValues>({
+    errors,
+    label,
+    name,
+    required,
+    children,
+}: Properties<T>): JSX.Element => {
+    const error = errors[name]?.message;
+    const hasError = Boolean(error);
+
+    return (
+        <View>
+            {label && (
+                <Text category="Label" style={globalStyles.mv5}>
+                    {label}
+                    {required && (
+                        <Text category="H6" style={styles.requiredFlag}>
+                            *
+                        </Text>
+                    )}
+                </Text>
+            )}
+
+            {React.Children.map(children, (child) =>
+                React.cloneElement(child as React.ReactElement<Properties<T>>, {
+                    hasError,
+                }),
+            )}
+
+            {hasError && (
+                <Text style={styles.errorText}>{error as string}</Text>
```

Comment:
```suggestion
                <Text style={styles.errorText}>{error}</Text>
```
There is already a check in `hasError`, so there is no need in `as string `

Reply:
When I delete `as string` it creates an error
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/104503014/922a82fb-262a-48a6-bdb9-f8cfe9abfe5a)

Reply:
Let's add check `{!!error && (...`